lms/security.py

```diff
@@ -28,34 +28,54 @@ class Permissions(Enum):
     ADMIN = "admin"
 
 
+class PathPolicyGetter:
+    """Callable that returns different policies based on the request path."""
+
+    def __init__(self, path_policy_mapping):
+        """
+        Create a new PathPolicyGetter.
+
+        :param path_policy_mapping: Map between route paths and policies
+        """
+        self.path_policy_mapping = path_policy_mapping
+
+    @lru_cache(maxsize=1)
+    def __call__(self, request):
+        """Get the corresponding policy based on the request path."""
+        for path, policy in self.path_policy_mapping.items():
+            # We rely on dict's preserved insertion order
+            # and expect that more specific routes are found first.
+            if request.path.startswith(path):
+                return policy()
+
+        raise ValueError(f"No policy found for {request.path}")
```

Comment:
This is great :+1: very easy to understand.

If `path_policy_mapping` were hard-coded rather than injectable then it could be hard-coded right here in this method rather than an `__init__()` and then everything would be easier to see all in one place. In fact the `PathPolicyGetter` class could possibly be replaced with just a function?